src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -109,16 +118,21 @@ async ValueTask SynchronizeSolutionAssetsWorkerAsync()
                 static (checksum, asset, checksumToObjects) => checksumToObjects.Add(checksum, asset),
                 arg: checksumToObjects, cancellationToken).ConfigureAwait(false);
 
+            using var _3 = ArrayBuilder<ProjectStateChecksums>.GetInstance(out var allProjectStateChecksums);
+
             // fourth, get all projects and documents in the solution 
             foreach (var (projectChecksum, _) in stateChecksums.Projects)
             {
                 var projectStateChecksums = (ProjectStateChecksums)checksumToObjects[projectChecksum];
-                await SynchronizeProjectAssetsAsync(projectStateChecksums, cancellationToken).ConfigureAwait(false);
+                allProjectStateChecksums.Add(projectStateChecksums);
             }
+
+            await SynchronizeProjectAssetsAsync(allProjectStateChecksums, cancellationToken).ConfigureAwait(false);
```

Comment:
pass in all projects of interest here so we can sync individual bits of data for all of them at once.